tests/specs/mcd/verification.k

```diff
@@ -130,6 +125,33 @@ module LEMMAS-MCD-HASKELL [symbolic, kore]
        andBool Y <=Int 0
        andBool notBool #rangeUInt(256, X -Int Y) [simplification]
 
+    rule { chop(X) #Equals chop(Y) } => { X #Equals Y }
+         requires #rangeSInt(256, X) andBool #rangeSInt(256, Y)
+         [simplification]
+
+    rule { chop ( X:Int ) #Equals chop ( Y:Int ) } =>
+         { true #Equals X ==Int Y andBool Y <=Int 0 }
+         requires (minSInt256 <=Int X) andBool (X <=Int 0) andBool #rangeSInt(256, Y) [simplification]
+
+  // ########################
+  // Buffer Reasoning
+  // ########################
+
+    rule { BA1:ByteArray ++ BA2:ByteArray #Equals BA1:ByteArray ++ BA3:ByteArray } => { BA2 #Equals BA3 } [simplification]
+
+    rule { BA1:ByteArray ++ (BA2:ByteArray ++ BA3:ByteArray) #Equals BA4:ByteArray ++ BA3 } => { (BA1 ++ BA2) #Equals BA4 } [simplification]
+
+    rule { #buf(32, BY:Int):ByteArray #Equals #buf(32, BZ:Int):ByteArray } => { BY #Equals BZ }
+         requires #rangeUInt(256, BY) andBool #rangeUInt(256, BZ) [simplification]
+
+    rule { #buf(32, BY:Int):ByteArray #Equals #buf(32, BZ:Int):ByteArray ++ BA2:ByteArray } =>
+         { true #Equals (BY ==Int BZ) andBool (.ByteArray ==K BA2) }
+         requires #rangeUInt(256, BY) andBool #rangeUInt(256, BZ) [simplification]
```

Comment:
We shouldn't need this lemma. If we can show `BA2 ==K .ByteArray`, we should instead just have a lemma which says:

```
rule BA ++ .ByteArray => BA [simplification]
```

To get rid of the unit bytearray.

Reply:
In the context in which this simplification is applied, we cannot show that `BA2 ==K .ByteArray`. I would like to keep this simplification as-is, and I give more context in the next comment below.